app/routes/endemics/date-of-visit-ms.js

```diff
@@ -0,0 +1,405 @@
+const Joi = require('joi')
+const { getReviewWithinLast10Months, getReviewTestResultWithinLast10Months } = require('../../api-requests/claim-service-api')
+const { dateOfVetVisitExceptions, claimType, livestockTypes } = require('../../constants/claim')
+const { labels } = require('../../config/visit-date')
+const session = require('../../session')
+const { endemicsClaim: { reviewTestResults: reviewTestResultsKey } } = require('../../session/keys')
+const raiseInvalidDataEvent = require('../../event/raise-invalid-data-event')
+const config = require('../../config')
+const urlPrefix = require('../../config').urlPrefix
+const { optionalPIHunt } = require('../../config')
+const {
+  endemicsDateOfVisit,
+  endemicsDateOfVisitException,
+  endemicsDateOfTesting,
+  endemicsSpeciesNumbers,
+  endemicsWhichTypeOfReview,
+  endemicsVetVisitsReviewTestResults
+} = require('../../config/routes')
+const {
+  endemicsClaim: {
+    dateOfVisit: dateOfVisitKey,
+    relevantReviewForEndemics: relevantReviewForEndemicsKey
+  }
+} = require('../../session/keys')
+const validateDateInputDay = require('../govuk-components/validate-date-input-day')
+const validateDateInputMonth = require('../govuk-components/validate-date-input-month')
+const validateDateInputYear = require('../govuk-components/validate-date-input-year')
+const { addError } = require('../utils/validations')
+const { getReviewType } = require('../../lib/get-review-type')
+const { getLivestockTypes } = require('../../lib/get-livestock-types')
+const appInsights = require('applicationinsights')
+const { canMakeReviewClaim, canMakeEndemicsClaim } = require('../../lib/can-make-claim')
+
+const pageUrl = `${urlPrefix}/${endemicsDateOfVisit}`
+
+const previousPageUrl = (latestVetVisitApplication, typeOfReview, previousClaims, typeOfLivestock) => {
+  const relevantClaims = previousClaims.filter(claim => claim.data.typeOfLivestock === typeOfLivestock)
+
+  const oldWorldClaimTypeOfLivestock = latestVetVisitApplication?.data?.whichReview
+
+  const isFirstTimeEndemicClaimForActiveOldWorldReviewClaim =
+    typeOfReview === claimType.endemics &&
+    [livestockTypes.beef, livestockTypes.dairy].includes(oldWorldClaimTypeOfLivestock) &&
+    relevantClaims.length === 0
+
+  if (isFirstTimeEndemicClaimForActiveOldWorldReviewClaim) { return `${urlPrefix}/${endemicsVetVisitsReviewTestResults}` }
+
+  return `${urlPrefix}/${endemicsWhichTypeOfReview}`
+}
+
+const isValidDateInput = (request, reviewOrFollowUpText) => {
```

Comment:
Can this validation be pulled out to somewhere separate? Believe this is a repeat of the validation in the non-ms version of this handler, which would make it an ideal candidate to be pulled out anyway, but regardless, might also make the handler easier to parse as a human without this embedded in here?

Reply:
Yeah for sure - me and Ross discussed that it looks horribly overengineered anyway and should maybe be simplified. Its just checking they put in valid numbers for the date, but its so long.

Reply:
Tried to write a simpler validation function but because of how we map the errors to the UI, its just too complicated to do as part of this ticket. Ive instead just imported the isValidDateInput from the non-MS file.

Reply:
And it turns out we cant do that either, because the isValidDateInput is slightly different in our new MS file (see the back link function) - I'll see if it can work